packages/odf/components/mcg/replication-policy-form.tsx

```diff
@@ -75,13 +89,37 @@ export const ReplicationPolicyForm: React.FC<ReplicationFormProps> = ({
     updateParentState(rules);
   };
 
+  const handleSyncDeletion = (value: boolean, ruleId: number) => {
+    const newRules = [...rules];
+    const index = newRules.findIndex((rule) => rule.id === ruleId);
+    newRules[index].syncDeletion = value;
+    setRules(newRules);
+    let temp = { ...logReplicationInfo };
+    if (!rules.some((rule) => rule.syncDeletion)) {
```

Comment:
`setRules(newRules)` will not update `rules` instantly...
```suggestion
    if (!rules.some((rule) => rule.syncDeletion) && value === false) {
```

Reply:
also, we have same check `!rules.some((rule) => rule.syncDeletion) ` at multiple places... do we need at all the places ??

Reply:
`setRules(newRules) will not update rules instantly...`  -> i did not see any issues with this ? what do you mean ?
![Screenshot from 2023-02-27 20-01-34](https://user-images.githubusercontent.com/57935785/221591189-11e51f45-bb2a-44fa-8532-49f15b19a3bd.png)

Reply:
`also, we have same check !rules.some((rule) => rule.syncDeletion)  at multiple places... do we need at all the places ??` -> yeah this condition is required in lot of places if even a single sync deletion box is checked, we need the log replication form inputs as well.

Reply:
u are setting "setRules(newRules)" on Line#96 then using "rules" on Line#98,
but "rules" on Line#98 will not have the changes u did on Line#96, until component re-renders again...

Reply:
either use `newRules.some ...` or `!rules.some((rule) => rule.syncDeletion) && value === false`

Reply:
ah ok, thanks for clarification.